src/OrleansRuntime/Silo/TestHooks/TestHooksSystemTarget.cs

```diff
@@ -37,6 +40,37 @@ public Task<string> GetConsistentRingProviderDiagnosticInfo()
 
         public Task<Guid> GetServiceId() => Task.FromResult(silo.GlobalConfig.ServiceId);
 
+        public Task<bool> CanGetStorageProvider(string providerName)
+        {
+            IStorageProvider tmp;
+            return Task.FromResult(silo.StorageProviderManager.TryGetProvider(providerName, out tmp));
+        }
+
+        public Task<bool> CanGetStreamProvider(string providerName)
+        {
+            try
+            {
+                silo.StreamProviderManager.GetStreamProvider(providerName);
+                return Task.FromResult(true);
+            }
+            catch (KeyNotFoundException)
+            {
+                return Task.FromResult(false);
+            }
+        }
+
+        public Task<bool> CanGetBoostraperProvider(string providerName)
+        {
+            foreach (var provider in silo.BootstrapProviders)
+            {
+                if (String.Equals(providerName, provider.Name))
+                {
+                    return Task.FromResult(true);
+                }
+            }
+            return Task.FromResult(false);
```

Comment:
yes the API are different. I didn't want to change the public Orleans API for this.